cloud/ec2_util.go

```diff
@@ -215,6 +223,129 @@ func timeTilNextHourlyPayment(host *host.Host) time.Duration {
 	return nextPaymentTime.Sub(now)
 }
 
+func getEc2SSHOptions(h *host.Host, keyName string) ([]string, error) {
+	if keyName == "" {
+		return nil, errors.New("No key specified for EC2 host")
+	}
+	opts := []string{"-i", keyName}
+	hasKnownHostsFile := false
+
+	for _, opt := range h.Distro.SSHOptions {
+		opt = strings.Trim(opt, " \t")
+		opts = append(opts, "-o", opt)
+		if strings.HasPrefix(opt, "UserKnownHostsFile") {
+			hasKnownHostsFile = true
+		}
+	}
+
+	if !hasKnownHostsFile {
+		opts = append(opts, "-o", "UserKnownHostsFile=/dev/null")
+	}
+	return opts, nil
+}
+
+func getKey(ctx context.Context, client AWSClient, credentials *credentials.Credentials, h *host.Host) (string, error) {
```

Comment:
I think I'd make the next three methods on the client.

Reply:
done.